### PR TITLE
WebexActivity Component

### PR DESCRIPTION
**Pull request by akoushke**
ready for review

### CODE REVIEW

**Comment by akoushke** [comment]
fix the README

**Comment by akoushke** [comment]
test the build

File: .storybook/setupTests.js [hunk 1]
```diff
@@ -10,3 +10,13 @@ global.shallow = shallow;
 global.render = render;
 global.mount = mount;
 global.rxjs = rxjs;
+global.Date = class extends Date {
```
Comment (adamweeks) [inline .storybook/setupTests.js]: Nice 👍

File: src/components/WebexActivity/WebexActivity.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  .content {
+    display: flex;
+    align-items: flex-start;
+    box-sizing: border-box;
+    min-width: 15.625rem !important;
+    font-size: 1rem;
+    color: $black;
+    word-wrap: break-word;
+    white-space: pre-wrap;
+    margin-left: 3.5rem; //avatar width + margin
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.scss]: You should make this a variable, since the author class also uses it.

File: src/components/WebexActivity/WebexActivity.stories.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+// Setup for the stories
+const [activityID] = Object.keys(activities);
+const stories = storiesOf('Webex Activity', module);
+const newActivities = {};
+
+// Stories
+stories.add('default', () => (
+  <WebexActivity
+    activityID={activityID}
+    adapters={{activitiesAdapter: new ActivitiesJSONAdapter(activities), peopleAdapter: new PeopleJSONAdapter(people)}}
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.stories.js]: Why isn't the style of this object broken down like the rest?

File: src/components/WebexActivity/WebexActivity.stories.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  />
+));
+
+stories.add('multi-line text', () => {
+  newActivities[activityID] = {...activities[activityID], text: 'line number1\nline number2\nline number3'};
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.stories.js]: We should have some "Lorem ipsum" text here

File: src/components/WebexActivity/WebexActivity.stories.js [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+      activityID={activityID}
+      adapters={{
+        activitiesAdapter: new ActivitiesJSONAdapter(newActivities),
+        peopleAdapter: new PeopleJSONAdapter(people),
+      }}
+    />
+  );
+});
+
+stories.add('multiple activities', () => (
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.stories.js]: I think this is something that belong with the Activity Stream and not with activity

File: src/components/WebexActivity/WebexActivity.stories.js [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+      activityID={activityID}
+      adapters={{
+        activitiesAdapter: new ActivitiesJSONAdapter(activities),
+        peopleAdapter: new PeopleJSONAdapter(people),
+      }}
+    />
+  </Fragment>
+));
+
+stories.add('instant multiple activities', () => {
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.stories.js]: I think this is something that belong with the Activity Stream and not with activity

File: src/components/WebexActivity/WebexActivity.stories.js [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        adapters={{
+          activitiesAdapter: new ActivitiesJSONAdapter(newActivities),
+          peopleAdapter: new PeopleJSONAdapter(people),
+        }}
+      />
+    </Fragment>
+  );
+});
+
+stories.add('today text', () => {
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.stories.js]: `created today`

File: src/components/WebexActivity/WebexActivity.stories.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+      activityID={activityID}
+      adapters={{
+        activitiesAdapter: new ActivitiesJSONAdapter(newActivities),
+        peopleAdapter: new PeopleJSONAdapter(people),
+      }}
+    />
+  );
+});
+
+stories.add('yesterday text', () => {
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.stories.js]: `created yesterday`

File: src/components/WebexActivity/WebexActivity.stories.js [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+      activityID={activityID}
+      adapters={{
+        activitiesAdapter: new ActivitiesJSONAdapter(newActivities),
+        peopleAdapter: new PeopleJSONAdapter(people),
+      }}
+    />
+  );
+});
+
+stories.add('same week text', () => {
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.stories.js]: `created this week`

File: src/components/WebexActivity/WebexActivity.stories.js [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+      activityID={activityID}
+      adapters={{
+        activitiesAdapter: new ActivitiesJSONAdapter(newActivities),
+        peopleAdapter: new PeopleJSONAdapter(people),
+      }}
+    />
+  );
+});
+
+stories.add('old text', () => {
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.stories.js]: `created over a week ago`

File: src/components/WebexAvatar/README.md [hunk 11]
```diff
@@ -1,26 +1,31 @@
...
 
 ## Embed
+
 1. Create a component adapter from which the data will be retrieved (See [adapters](../../adapters)). For instance:
 
-    ```js
-      const jsonAdapter = new PeopleJSONAdapter();
-    ```
+   ```js
+   const adapter = new PeopleJSONAdapter(people);
```
Comment (lalli-flores) [inline src/components/WebexAvatar/README.md]: don't change the variable name, I think is better to have it explicitly say "json" as its type

File: src/components/WebexAvatar/WebexAvatar.test.js [hunk 12]
```diff
@@ -9,86 +9,97 @@ import WebexAvatar from './WebexAvatar';
...
 describe('Webex Avatar component', () => {
-  let personID, peopleJSONAdapter;
+  let personID, newPeople, peopleAdapter;
 
   beforeEach(() => {
     [personID] = Object.keys(people);
-    peopleJSONAdapter = new PeopleJSONAdapter(people);
-    global.console.error = jest.fn();
+    newPeople = people;
+    peopleAdapter = new PeopleJSONAdapter(people);
```
Comment (lalli-flores) [inline src/components/WebexAvatar/WebexAvatar.test.js]: why did you change the variable name?

File: src/components/hooks/usePerson.js [hunk 13]
```diff
@@ -8,11 +8,12 @@ import {useState, useEffect} from 'react';
  * @returns {Person} Data of the person
  */
 export default function usePerson(personID, adapter) {
-  const [person, setPerson] = useState(undefined);
+  const [person, setPerson] = useState({});
```
Comment (lalli-flores) [inline src/components/hooks/usePerson.js]: Why did you change this back?

File: .storybook/config.js [hunk 14]
```diff
@@ -1,9 +1,8 @@
-import { configure } from '@storybook/react';
-import '../momentum-ui.scss';
```
[the remark this answers is not in the document]
Comment (akoushke) [reply .storybook/config.js]: no

File: src/components/hooks/usePerson.js [hunk 15]
```diff
@@ -8,11 +8,12 @@ import {useState, useEffect} from 'react';
...
 
   useEffect(() => {
-    // eslint-disable-next-line no-console
-    const onError = (error) => console.error(error.message);
+    const onError = (error) => {
+      throw new Error(error.message);
```
[the remark this answers is not in the document]
Comment (akoushke) [reply src/components/hooks/usePerson.js]: I thought throwing error is more user friendly than a console error.

[the remark this answers is not in the document]
Comment (lalli-flores) [reply src/components/hooks/usePerson.js]: We shouldn't be crashing the "application" if something goes wrong, but this is fine until we have error states.

File: src/components/WebexActivity/WebexActivity.stories.js (hunk 3 above)
Comment (akoushke) [reply src/components/WebexActivity/WebexActivity.stories.js]: its shorter

File: src/components/hooks/usePerson.js (hunk 13 above)
Comment (akoushke) [reply src/components/hooks/usePerson.js]: since we are not doing nested object, we dont have to validate the states anymore.

File: src/components/WebexAvatar/WebexAvatar.test.js (hunk 12 above)
Comment (akoushke) [reply src/components/WebexAvatar/WebexAvatar.test.js]: to be consistent with `WebexActivity.test.js` naming. and I can't change that to `peopleJSONAdapter`, since it's defined at the component level. Adapters prop in webexActivty is an object that contains two adapters.

File: src/components/WebexActivity/WebexActivity.js [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+    </div>
+  );
+}
+
+WebexActivity.propTypes = {
+  activityID: PropTypes.string.isRequired,
+  adapters: PropTypes.exact({
+    activitiesAdapter: PropTypes.object.isRequired,
+    peopleAdapter: PropTypes.object.isRequired,
+  }),
```
Comment (adamweeks) [inline src/components/WebexActivity/WebexActivity.js]: Are you able to add `.isRequired` here and remove the `defaultProps` definition?

File: src/components/WebexActivity/README.md [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+1. Create a component adapters object from which the data will be retrieved (See [adapters](../../adapters)). For instance:
+
+   ```js
+   const jsonAdapters = {
+     peopleJSONAdapter: new PeopleJSONAdapter(people),
+     activitiesJSONAdapter: new ActivitiesJSONAdapter(activities),
+   };
+   ```
+
+2. Create a component instance by passing the activity ID as a string and the component data adapters which contains [PeopleAdapter](../../adapters/PeopleAdapter.js) and [ActivitiesAdapter](../../adapters/ActivitiesAdapter.js)that we created previously
```
Comment (lalli-flores) [inline src/components/WebexActivity/README.md]: A space is needed after the `ActivitiesAdapter` link (See the README "view")

File: src/components/WebexActivity/WebexActivity.js [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+
+import WebexAvatar from '../WebexAvatar/WebexAvatar';
+import {useActivity, usePerson} from '../hooks';
+
+import './WebexActivity.scss';
+
+/**
+ * Returns a formatted timestamp based on the given date's offset from the current time.
+ *
+ * @param {Date} timestamp A string value representing a date, specified in a format recognized by the Date.parse()
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.js]: I had this param description for before, when it was a string... now it's a date object 🤦🏻‍♀️
Maybe something like `Date instance to format` ??

File: src/components/WebexActivity/WebexActivity.js (hunk 16 above)
Comment (akoushke) [reply src/components/WebexActivity/WebexActivity.js]: yeah that works

File: src/components/WebexActivity/WebexActivity.js [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+export function Header(props) {
+  const {personID, adapter, created} = props;
+  const {displayName} = usePerson(personID, adapter);
+
+  return (
+    <div className="activity-header">
+      <WebexAvatar personID={personID} adapter={adapter} />
+      <div className="activity-author">
+        <span>{displayName}</span>
+        <span className="activity-timestamp">{renderedDate(new Date(created))}</span>
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.js]: yeehh 🙌🏻

File: src/components/WebexActivity/WebexActivity.test.js [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+    test('matches snapshot with "default" props', () => {
+      expect(
+        shallow(
+          <Header personID={personID} adapter={adapters.peopleAdapter} created={activities[activityID].created} />
+        )
+      ).toMatchSnapshot();
+    });
+  });
+
+  describe('snapshots', () => {
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.test.js]: `Webex Activity snapshots`

File: src/components/WebexActivity/WebexActivity.test.js [hunk 21]
```diff
@@ -0,0 +1,123 @@
...
+    test('matches snapshot with a text from the same week', () => {
+      newActivities[activityID] = {
+        ...activities[activityID],
+        created: 'sameWeek',
+      };
+      adapters.activitiesAdapter = new ActivitiesJSONAdapter(newActivities);
+
+      expect(shallow(<WebexActivity activityID={activityID} adapters={adapters} />)).toMatchSnapshot();
+    });
+  });
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.test.js]: There's a snapshot missing from "over a week ago"

File: src/components/hooks/useActivity.js [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+  useEffect(() => {
+    const onError = (error) => {
+      throw error;
+    };
+    const subscription = activitiesAdapter.getActivity(activityID).subscribe(setActivity, onError);
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [activity]);
```
Comment (lalli-flores) [inline src/components/hooks/useActivity.js]: This should be `[]`. 
You can see [my explanation in this PR](https://github.com/webex/components/pull/23)

File: src/components/WebexAvatar/WebexAvatar.js [hunk 23]
```diff
@@ -5,16 +5,10 @@ import {Avatar} from '@momentum-ui/react';
 import usePerson from '../hooks/usePerson';
 
 export default function WebexAvatar(props) {
-  let component = null;
-  const person = usePerson(props.personID, props.adapter);
+  const {personID, adapter} = props;
+  const {avatar, displayName, status} = usePerson(personID, adapter);
```
Comment (lalli-flores) [inline src/components/WebexAvatar/WebexAvatar.js]: Would
`const {avatar, displayName, status} = usePerson({personID, adapter});` work? and we'd get rid of the top definition?

Comment (akoushke) [reply src/components/WebexAvatar/WebexAvatar.js]: no, personID and adapter must be defined in order to be in the right hand side of the `=`

File: src/components/WebexActivity/WebexActivity.stories.js [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+
+import WebexActivity from './WebexActivity';
+
+// Setup for the stories
+const [activityID] = Object.keys(activities);
+const stories = storiesOf('Webex Activity', module);
+const newActivities = {};
+
+// Stories
+stories.add('default', () => (
```
Comment (lalli-flores) [inline src/components/WebexActivity/WebexActivity.stories.js]: You're missing a "no header" story

Comment (akoushke) [reply src/components/WebexActivity/WebexActivity.stories.js]: I added one
